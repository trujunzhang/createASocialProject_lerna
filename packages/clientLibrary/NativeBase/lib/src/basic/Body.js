import React, { Component } from "react";
import * as PropTypes from "prop-types";
import { View, ViewPropTypes } from "react-primitives";

import { connectStyle } from "@app/native-base-shoutem-theme";
import mapPropsToStyleNames from "../utils/mapPropsToStyleNames";

class Body extends Component {
  render() {
    return <View ref={c => (this._root = c)} {...this.props} />;
  }
}

Body.propTypes = {
  ...ViewPropTypes,
  style: PropTypes.oneOfType([
    PropTypes.object,
    PropTypes.number,
    PropTypes.array
  ])
};

const StyledBody = connectStyle("NativeBase.Body", {}, mapPropsToStyleNames)(
  Body
);

export { StyledBody as Body };
