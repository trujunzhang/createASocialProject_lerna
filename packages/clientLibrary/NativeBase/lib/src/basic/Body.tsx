import * as React from 'react'
import PropTypes from "prop-types";
import { View, ViewPropTypes } from "react-primitives";

import { connectStyle } from "@app/native-base-shoutem-theme";
import mapPropsToStyleNames from "../utils/mapPropsToStyleNames";

class Body extends React.Component {
  private _root: any
  render() {
    return <View ref={c => (this._root = c)} {...this.props} />;
  }
}

// Body.propTypes = {
//   ...ViewPropTypes,
//   style: PropTypes.oneOfType([
//     PropTypes.object,
//     PropTypes.number,
//     PropTypes.array
//   ])
// };

const StyledBody = connectStyle("NativeBase.Body", {}, mapPropsToStyleNames)(
  Body
);

export { StyledBody as Body };
