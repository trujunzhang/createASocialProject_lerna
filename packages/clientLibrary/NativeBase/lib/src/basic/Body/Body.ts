import * as React from 'react'

import { ReactNative, RnViewStyleProp } from '@appUtils/rnCommonProps'

export interface BodyProps {
  style?: RnViewStyleProp | Array<RnViewStyleProp>
}
