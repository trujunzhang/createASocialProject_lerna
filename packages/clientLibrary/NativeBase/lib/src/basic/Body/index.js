import * as React from 'react';
import { View } from 'react-primitives';
import { connectStyle } from 'native-base-shoutem-theme';
import { mapPropsToStyleNames } from '@app/tools';
class Body extends React.Component {
    render() {
        return React.createElement(View, Object.assign({ name: 'body', ref: (c) => (this._root = c) }, this.props));
    }
}
// Body.propTypes = {
//   ...ViewPropTypes,
//   style: PropTypes.oneOfType([PropTypes.object, PropTypes.number, PropTypes.array])
// }
const StyledBody = connectStyle('NativeBase.Body', {}, mapPropsToStyleNames)(Body);
export { StyledBody as Body };
//# sourceMappingURL=index.js.map