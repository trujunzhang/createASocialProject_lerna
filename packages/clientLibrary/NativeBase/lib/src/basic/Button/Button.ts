import * as React from 'react'

import { ReactNative, RnViewStyleProp, RnTextStyleProp, BsStyle } from '@appUtils/rnCommonProps'

/**
 * see Widget Button.js
 */
export interface ButtonProps extends ReactNative.TouchableOpacityProps, BsStyle {
  /**
   * Defines button style
   */
  style?: RnViewStyleProp | Array<RnViewStyleProp>
  /**
   * Defines button text style
   */
  textStyle?: RnTextStyleProp
  // onPress?: () => void
  /**
   * Block level button
   */
  block?: boolean
  /**
   * Vertical button
   */
  vertical?: boolean
  /**
   * Button with badges
   */
  badge?: boolean
  //primary?: boolean,
  /**
   * Gives you effect of Icon-buttons.
   * To have button with transparent background, include this prop.
   */
  transparent?: boolean
  //success?: boolean,
  //danger?: boolean,
  // warning?: boolean,
  //info?: boolean,
  color?: string
  /**
   * Applies outline button style.
   */
  bordered?: boolean
  /**
   * Renders button with slightly round shaped edges.
   */
  rounded?: boolean
  /**
   * For large size button
   */
  large?: boolean
  /**
   * For small size button
   */
  small?: boolean
  /**
   * Used for Icon alignment.
   * Aligns icon to the left in button.
   * By default, icons are aligned to the left in button.
   */
  iconLeft?: boolean
  /**
   * Used for Icon alignment.
   * Aligns icon to the right in button.
   */
  iconRight?: boolean
  /**
   * Used to enable/disable icons
   * Icons align in the center of the button.
   */
  icon?: boolean
  /**
   * Disables onPress option for button
   */
  disabled?: boolean
  active?: boolean
  inputButton?: boolean
  full?: boolean
  light?: boolean
  dark?: boolean
  /**
   * [android] colored ripple effect
   */
  androidRippleColor?: string
  /**
   * Segments
   */
  first?: boolean
  last?: boolean
  /**
   * Adds necessary padding when Text button defined in Left / Right of Header (iOS)
   */
  hasText?: boolean
  // Others
  activeOpacity?: any
  onlyCenter?: boolean
  facebook?: boolean
  twitter?: boolean
  svgLeft?: boolean
}
