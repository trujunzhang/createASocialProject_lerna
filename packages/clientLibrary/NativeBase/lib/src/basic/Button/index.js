import * as React from 'react';
import * as PropTypes from 'prop-types';
import { TouchableOpacity, Platform, View, TouchableNativeFeedback, StyleSheet } from 'react-primitives';
import { connectStyle } from 'native-base-shoutem-theme';
import { platformVariables as variable } from '@appVariables/theme/variables/platform'; // variables // [variable]
import { Text } from '@appBasic/Text';
import computeProps from '@appUtils/computeProps';
import { mapPropsToStyleNames } from '@app/tools';
class Button extends React.Component {
    getInitialStyle() {
        return {
            borderedBtn: {
                borderWidth: this.props.bordered ? 1 : undefined,
                borderRadius: this.props.rounded && this.props.bordered ? variable.borderRadiusLarge : 2
            }
        };
    }
    prepareRootProps() {
        const defaultProps = {
            style: this.getInitialStyle().borderedBtn
        };
        if (Array.isArray(this.props.style)) {
            const flattenedStyle = this.props.style.reduce((accumulator, currentValue) => accumulator.concat(currentValue), []);
            return computeProps({ ...this.props, style: flattenedStyle }, defaultProps);
        }
        return computeProps(this.props, defaultProps);
    }
    render() {
        const variables = this.context.theme
            ? this.context.theme['@@shoutem.theme/themeStyle'].variables
            : variable;
        const children = Platform.OS === 'ios'
            ? this.props.children
            : React.Children.map(this.props.children, (child) => child && child.type === Text
                ? React.cloneElement(child, {
                    uppercase: variables.btnUppercaseAndroidText,
                    ...child.props
                })
                : child);
        if (Platform.OS === 'ios' ||
            Platform.OS === 'web' ||
            variables.androidRipple === false ||
            Platform['Version'] < 21) {
            return (React.createElement(TouchableOpacity, Object.assign({}, this.prepareRootProps(), { ref: (c) => (this._root = c), activeOpacity: this.props.activeOpacity > 0 ? this.props.activeOpacity : 0.5 }), children));
        }
        else {
            if (this.props.rounded) {
                let buttonStyle = { ...this.prepareRootProps().style };
                let buttonFlex = this.props.full || this.props.block ? 1 : buttonStyle.flex;
                return (React.createElement(View, { style: [
                        { maxHeight: buttonStyle.height },
                        buttonStyle,
                        { paddingTop: undefined, paddingBottom: undefined }
                    ] },
                    React.createElement(TouchableNativeFeedback, Object.assign({ ref: (c) => (this._root = c), background: this.props.androidRippleColor
                            ? TouchableNativeFeedback.Ripple(this.props.androidRippleColor, true)
                            : TouchableNativeFeedback.Ripple(variables.androidRippleColor, true) }, this.prepareRootProps()),
                        React.createElement(View, { style: [
                                styles.childContainer,
                                {
                                    paddingTop: buttonStyle.paddingTop,
                                    paddingBottom: buttonStyle.paddingBottom,
                                    height: buttonStyle.height,
                                    flexGrow: buttonFlex
                                }
                            ] }, children))));
            }
            else {
                return (React.createElement(TouchableNativeFeedback, Object.assign({ ref: (c) => (this._root = c), onPress: this.props.onPress, background: this.props.transparent
                        ? TouchableNativeFeedback.Ripple('transparent')
                        : TouchableNativeFeedback.Ripple(variables.androidRippleColor, false) }, this.prepareRootProps()),
                    React.createElement(View, Object.assign({}, this.prepareRootProps()), children)));
            }
        }
    }
}
Button.contextTypes = {
    theme: PropTypes.object
};
// Button.propTypes = {
//   ...TouchableOpacity.propTypes,
//   style: PropTypes.oneOfType([
//     PropTypes.object,
//     PropTypes.number,
//     PropTypes.array
//   ]),
//   block: PropTypes.bool,
//   primary: PropTypes.bool,
//   transparent: PropTypes.bool,
//   success: PropTypes.bool,
//   danger: PropTypes.bool,
//   warning: PropTypes.bool,
//   info: PropTypes.bool,
//   bordered: PropTypes.bool,
//   disabled: PropTypes.bool,
//   rounded: PropTypes.bool,
//   large: PropTypes.bool,
//   small: PropTypes.bool,
//   active: PropTypes.bool
// };
const styles = StyleSheet.create({
    childContainer: {
        flexShrink: 1,
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center'
    }
});
const StyledButton = connectStyle('NativeBase.Button', {}, mapPropsToStyleNames)(Button);
export { StyledButton as Button };
//# sourceMappingURL=index.js.map