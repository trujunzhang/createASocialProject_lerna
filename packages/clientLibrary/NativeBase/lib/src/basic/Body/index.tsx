import * as React from 'react'
import * as PropTypes from 'prop-types'
import { View } from 'react-primitives'

import { connectStyle } from 'native-base-shoutem-theme'
import { mapPropsToStyleNames } from '@app/tools'

import { BodyProps } from './Body'

class Body extends React.Component<BodyProps, any> {
  private _root: any
  render() {
    return <View name={'body'} ref={(c) => (this._root = c)} {...this.props} />
  }
}

// Body.propTypes = {
//   ...ViewPropTypes,
//   style: PropTypes.oneOfType([PropTypes.object, PropTypes.number, PropTypes.array])
// }

const StyledBody = connectStyle('NativeBase.Body', {}, mapPropsToStyleNames)(Body)

export { StyledBody as Body }
