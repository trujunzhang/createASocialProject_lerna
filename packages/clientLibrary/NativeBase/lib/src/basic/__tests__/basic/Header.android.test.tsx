import 'react-primitives'
import * as React from 'react'
import * as renderer from 'react-test-renderer'
import { Container } from '../../Container'
import { Header } from '../../Header'
import { Left } from '../../Left'
import { Button } from '../../Button'
import { Icon } from '../../Icon'
import { Body } from '../../Body'
import { Title } from '../../Title'
import { Subtitle } from '../../Subtitle'
import { Right } from '../../Right'
import { Content } from '../../Content'
import { Text } from '../../Text'

// Note: test renderer must be required after react-native.

// jest.mock("Platform", () => {
//   const Platform = require.requireActual("Platform");
//   Platform.OS = "android";
//  return Platform;
// });

it('renders Header with only Title', () => {
  const tree = renderer
    .create(
      <Container>
        <Header>
          <Left />
          <Body>
            <Title>Header</Title>
          </Body>
          <Right />
        </Header>
      </Container>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders Header with Icon Buttons', () => {
  const tree = renderer
    .create(
      <Container>
        <Header>
          <Left>
            <Button transparent>
              <Icon name="arrow-back" />
            </Button>
          </Left>
          <Body>
            <Title>Header</Title>
          </Body>
          <Right>
            <Button transparent>
              <Icon name="menu" />
            </Button>
          </Right>
        </Header>
      </Container>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders Header with Text Buttons', () => {
  const tree = renderer
    .create(
      <Container>
        <Header>
          <Left>
            <Button hasText transparent>
              <Text>Back</Text>
            </Button>
          </Left>
          <Body>
            <Title>Header</Title>
          </Body>
          <Right>
            <Button hasText transparent>
              <Text>Cancel</Text>
            </Button>
          </Right>
        </Header>
      </Container>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders Header with Icon Button & Text Button', () => {
  const tree = renderer
    .create(
      <Container>
        <Header>
          <Left>
            <Button transparent>
              <Icon name="arrow-back" />
            </Button>
          </Left>
          <Body>
            <Title>Header</Title>
          </Body>
          <Right>
            <Button hasText transparent>
              <Text>Cancel</Text>
            </Button>
          </Right>
        </Header>
      </Container>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders Header with Icon & Text Button', () => {
  const tree = renderer
    .create(
      <Container>
        <Header>
          <Left>
            <Button transparent>
              <Icon name="arrow-back" />
              <Text>Back</Text>
            </Button>
          </Left>
          <Body>
            <Title>Header</Title>
          </Body>
          <Right>
            <Button hasText transparent>
              <Text>Cancel</Text>
            </Button>
          </Right>
        </Header>
      </Container>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders Header with multiple Icon Button', () => {
  const tree = renderer
    .create(
      <Container>
        <Header>
          <Left>
            <Button transparent>
              <Icon name="arrow-back" />
            </Button>
          </Left>
          <Body>
            <Title>Header</Title>
          </Body>
          <Right>
            <Button transparent>
              <Icon name="search" />
            </Button>
            <Button transparent>
              <Icon name="heart" />
            </Button>
            <Button transparent>
              <Icon name="more" />
            </Button>
          </Right>
        </Header>
      </Container>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders Header with Title and Subtitle', () => {
  const tree = renderer
    .create(
      <Container>
        <Header hasSubtitle>
          <Left>
            <Button transparent>
              <Icon name="arrow-back" />
            </Button>
          </Left>
          <Body>
            <Title>Title</Title>
            <Subtitle>Subtitle</Subtitle>
          </Body>
          <Right />
        </Header>
      </Container>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders Header with Custom background color', () => {
  const tree = renderer
    .create(
      <Container>
        <Header
          style={{ backgroundColor: '#dc4239' }}
          androidStatusBarColor="#dc2015"
          iosBarStyle="light-content">
          <Left>
            <Button transparent>
              <Icon name="arrow-back" style={{ color: '#FFF' }} />
            </Button>
          </Left>
          <Body>
            <Title style={{ color: '#FFF' }}>Custom Header</Title>
          </Body>
          <Right />
        </Header>
      </Container>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders Header span', () => {
  const tree = renderer
    .create(
      <Container>
        <Header span>
          <Left>
            <Button transparent>
              <Icon name="arrow-back" />
            </Button>
          </Left>
          <Body>
            <Title>Header Span</Title>
          </Body>
          <Right />
        </Header>
      </Container>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders Header noShadow', () => {
  const tree = renderer
    .create(
      <Container>
        <Header noShadow>
          <Left>
            <Button transparent>
              <Icon name="arrow-back" />
            </Button>
          </Left>
          <Body>
            <Title>Header No Shadow</Title>
          </Body>
          <Right>
            <Button transparent>
              <Icon name="menu" />
            </Button>
          </Right>
        </Header>
      </Container>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders Header noLeft', () => {
  const tree = renderer
    .create(
      <Container>
        <Header noLeft>
          <Left>
            <Button hasText transparent>
              <Text>Back</Text>
            </Button>
          </Left>
          <Body>
            <Title>Header</Title>
          </Body>
          <Right>
            <Button hasText transparent>
              <Text>Cancel</Text>
            </Button>
          </Right>
        </Header>
      </Container>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
