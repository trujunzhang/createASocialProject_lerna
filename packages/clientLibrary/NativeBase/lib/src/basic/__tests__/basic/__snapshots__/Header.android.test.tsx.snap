// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Header noLeft 1`] = `
<view
  name="container"
  style={Object {}}
>
  <view
    name="header"
  >
    <view
      name="statusBar"
    />
    <view
      name="header-other"
      style={Object {}}
    >
      <view
        name="left"
        style={Object {}}
      >
        <view
          name="View"
          style={
            Object {
              "borderBottomLeftRadius": 2,
              "borderBottomRightRadius": 2,
              "borderTopLeftRadius": 2,
              "borderTopRightRadius": 2,
            }
          }
        >
          <view
            name="View"
            style={Object {}}
          >
            <text
              style={Object {}}
            >
              Back
            </text>
          </view>
        </view>
      </view>
      <view
        name="body"
        style={Object {}}
      >
        <view
          name="View"
          style={Object {}}
        >
          <text
            style={Object {}}
          >
            Header
          </text>
        </view>
      </view>
      <view
        name="right"
        style={Object {}}
      >
        <view
          name="View"
          style={
            Object {
              "borderBottomLeftRadius": 2,
              "borderBottomRightRadius": 2,
              "borderTopLeftRadius": 2,
              "borderTopRightRadius": 2,
            }
          }
        >
          <view
            name="View"
            style={Object {}}
          >
            <text
              style={Object {}}
            >
              Cancel
            </text>
          </view>
        </view>
      </view>
    </view>
  </view>
</view>
`;

exports[`renders Header noShadow 1`] = `
<view
  name="container"
  style={Object {}}
>
  <view
    name="header"
  >
    <view
      name="statusBar"
    />
    <view
      name="header-other"
      style={Object {}}
    >
      <view
        name="left"
        style={Object {}}
      >
        <view
          name="View"
          style={
            Object {
              "borderBottomLeftRadius": 2,
              "borderBottomRightRadius": 2,
              "borderTopLeftRadius": 2,
              "borderTopRightRadius": 2,
            }
          }
        >
          <view
            name="View"
            style={Object {}}
          >
            <text
              style={Object {}}
            >
              ios-arrow-back
            </text>
          </view>
        </view>
      </view>
      <view
        name="body"
        style={Object {}}
      >
        <view
          name="View"
          style={Object {}}
        >
          <text
            style={Object {}}
          >
            Header No Shadow
          </text>
        </view>
      </view>
      <view
        name="right"
        style={Object {}}
      >
        <view
          name="View"
          style={
            Object {
              "borderBottomLeftRadius": 2,
              "borderBottomRightRadius": 2,
              "borderTopLeftRadius": 2,
              "borderTopRightRadius": 2,
            }
          }
        >
          <view
            name="View"
            style={Object {}}
          >
            <text
              style={Object {}}
            >
              ios-menu
            </text>
          </view>
        </view>
      </view>
    </view>
  </view>
</view>
`;

exports[`renders Header span 1`] = `
<view
  name="container"
  style={Object {}}
>
  <view
    name="header"
  >
    <view
      name="statusBar"
    />
    <view
      name="header-other"
      style={Object {}}
    >
      <view
        name="left"
        style={Object {}}
      >
        <view
          name="View"
          style={
            Object {
              "borderBottomLeftRadius": 2,
              "borderBottomRightRadius": 2,
              "borderTopLeftRadius": 2,
              "borderTopRightRadius": 2,
            }
          }
        >
          <view
            name="View"
            style={Object {}}
          >
            <text
              style={Object {}}
            >
              ios-arrow-back
            </text>
          </view>
        </view>
      </view>
      <view
        name="body"
        style={Object {}}
      >
        <view
          name="View"
          style={Object {}}
        >
          <text
            style={Object {}}
          >
            Header Span
          </text>
        </view>
      </view>
      <view
        name="right"
        style={Object {}}
      />
    </view>
  </view>
</view>
`;

exports[`renders Header with Custom background color 1`] = `
<view
  name="container"
  style={Object {}}
>
  <view
    name="header"
  >
    <view
      name="statusBar"
    />
    <view
      name="header-other"
      style={
        Object {
          "backgroundColor": "#dc4239",
        }
      }
    >
      <view
        name="left"
        style={Object {}}
      >
        <view
          name="View"
          style={
            Object {
              "borderBottomLeftRadius": 2,
              "borderBottomRightRadius": 2,
              "borderTopLeftRadius": 2,
              "borderTopRightRadius": 2,
              "color": "#FFF",
            }
          }
        >
          <view
            name="View"
            style={Object {}}
          >
            <text
              style={Object {}}
            >
              ios-arrow-back
            </text>
          </view>
        </view>
      </view>
      <view
        name="body"
        style={Object {}}
      >
        <view
          name="View"
          style={Object {}}
        >
          <text
            style={Object {}}
          >
            Custom Header
          </text>
        </view>
      </view>
      <view
        name="right"
        style={Object {}}
      />
    </view>
  </view>
</view>
`;

exports[`renders Header with Icon & Text Button 1`] = `
<view
  name="container"
  style={Object {}}
>
  <view
    name="header"
  >
    <view
      name="statusBar"
    />
    <view
      name="header-other"
      style={Object {}}
    >
      <view
        name="left"
        style={Object {}}
      >
        <view
          name="TouchableOpacity: "
        />
      </view>
      <view
        name="body"
        style={Object {}}
      >
        <view
          name="View"
          style={Object {}}
        >
          <text
            style={Object {}}
          >
            Header
          </text>
        </view>
      </view>
      <view
        name="right"
        style={Object {}}
      >
        <view
          name="View"
          style={
            Object {
              "borderBottomLeftRadius": 2,
              "borderBottomRightRadius": 2,
              "borderTopLeftRadius": 2,
              "borderTopRightRadius": 2,
            }
          }
        >
          <view
            name="View"
            style={Object {}}
          >
            <text
              style={Object {}}
            >
              Cancel
            </text>
          </view>
        </view>
      </view>
    </view>
  </view>
</view>
`;

exports[`renders Header with Icon Button & Text Button 1`] = `
<view
  name="container"
  style={Object {}}
>
  <view
    name="header"
  >
    <view
      name="statusBar"
    />
    <view
      name="header-other"
      style={Object {}}
    >
      <view
        name="left"
        style={Object {}}
      >
        <view
          name="View"
          style={
            Object {
              "borderBottomLeftRadius": 2,
              "borderBottomRightRadius": 2,
              "borderTopLeftRadius": 2,
              "borderTopRightRadius": 2,
            }
          }
        >
          <view
            name="View"
            style={Object {}}
          >
            <text
              style={Object {}}
            >
              ios-arrow-back
            </text>
          </view>
        </view>
      </view>
      <view
        name="body"
        style={Object {}}
      >
        <view
          name="View"
          style={Object {}}
        >
          <text
            style={Object {}}
          >
            Header
          </text>
        </view>
      </view>
      <view
        name="right"
        style={Object {}}
      >
        <view
          name="View"
          style={
            Object {
              "borderBottomLeftRadius": 2,
              "borderBottomRightRadius": 2,
              "borderTopLeftRadius": 2,
              "borderTopRightRadius": 2,
            }
          }
        >
          <view
            name="View"
            style={Object {}}
          >
            <text
              style={Object {}}
            >
              Cancel
            </text>
          </view>
        </view>
      </view>
    </view>
  </view>
</view>
`;

exports[`renders Header with Icon Buttons 1`] = `
<view
  name="container"
  style={Object {}}
>
  <view
    name="header"
  >
    <view
      name="statusBar"
    />
    <view
      name="header-other"
      style={Object {}}
    >
      <view
        name="left"
        style={Object {}}
      >
        <view
          name="View"
          style={
            Object {
              "borderBottomLeftRadius": 2,
              "borderBottomRightRadius": 2,
              "borderTopLeftRadius": 2,
              "borderTopRightRadius": 2,
            }
          }
        >
          <view
            name="View"
            style={Object {}}
          >
            <text
              style={Object {}}
            >
              ios-arrow-back
            </text>
          </view>
        </view>
      </view>
      <view
        name="body"
        style={Object {}}
      >
        <view
          name="View"
          style={Object {}}
        >
          <text
            style={Object {}}
          >
            Header
          </text>
        </view>
      </view>
      <view
        name="right"
        style={Object {}}
      >
        <view
          name="View"
          style={
            Object {
              "borderBottomLeftRadius": 2,
              "borderBottomRightRadius": 2,
              "borderTopLeftRadius": 2,
              "borderTopRightRadius": 2,
            }
          }
        >
          <view
            name="View"
            style={Object {}}
          >
            <text
              style={Object {}}
            >
              ios-menu
            </text>
          </view>
        </view>
      </view>
    </view>
  </view>
</view>
`;

exports[`renders Header with Text Buttons 1`] = `
<view
  name="container"
  style={Object {}}
>
  <view
    name="header"
  >
    <view
      name="statusBar"
    />
    <view
      name="header-other"
      style={Object {}}
    >
      <view
        name="left"
        style={Object {}}
      >
        <view
          name="View"
          style={
            Object {
              "borderBottomLeftRadius": 2,
              "borderBottomRightRadius": 2,
              "borderTopLeftRadius": 2,
              "borderTopRightRadius": 2,
            }
          }
        >
          <view
            name="View"
            style={Object {}}
          >
            <text
              style={Object {}}
            >
              Back
            </text>
          </view>
        </view>
      </view>
      <view
        name="body"
        style={Object {}}
      >
        <view
          name="View"
          style={Object {}}
        >
          <text
            style={Object {}}
          >
            Header
          </text>
        </view>
      </view>
      <view
        name="right"
        style={Object {}}
      >
        <view
          name="View"
          style={
            Object {
              "borderBottomLeftRadius": 2,
              "borderBottomRightRadius": 2,
              "borderTopLeftRadius": 2,
              "borderTopRightRadius": 2,
            }
          }
        >
          <view
            name="View"
            style={Object {}}
          >
            <text
              style={Object {}}
            >
              Cancel
            </text>
          </view>
        </view>
      </view>
    </view>
  </view>
</view>
`;

exports[`renders Header with Title and Subtitle 1`] = `
<view
  name="container"
  style={Object {}}
>
  <view
    name="header"
  >
    <view
      name="statusBar"
    />
    <view
      name="header-other"
      style={Object {}}
    >
      <view
        name="left"
        style={Object {}}
      >
        <view
          name="View"
          style={
            Object {
              "borderBottomLeftRadius": 2,
              "borderBottomRightRadius": 2,
              "borderTopLeftRadius": 2,
              "borderTopRightRadius": 2,
            }
          }
        >
          <view
            name="View"
            style={Object {}}
          >
            <text
              style={Object {}}
            >
              ios-arrow-back
            </text>
          </view>
        </view>
      </view>
      <view
        name="body"
        style={Object {}}
      >
        <view
          name="View"
          style={Object {}}
        >
          <text
            style={Object {}}
          >
            Title
          </text>
        </view>
        <view
          name="View"
          style={Object {}}
        >
          <text
            style={Object {}}
          >
            Subtitle
          </text>
        </view>
      </view>
      <view
        name="right"
        style={Object {}}
      />
    </view>
  </view>
</view>
`;

exports[`renders Header with multiple Icon Button 1`] = `
<view
  name="container"
  style={Object {}}
>
  <view
    name="header"
  >
    <view
      name="statusBar"
    />
    <view
      name="header-other"
      style={Object {}}
    >
      <view
        name="left"
        style={Object {}}
      >
        <view
          name="View"
          style={
            Object {
              "borderBottomLeftRadius": 2,
              "borderBottomRightRadius": 2,
              "borderTopLeftRadius": 2,
              "borderTopRightRadius": 2,
            }
          }
        >
          <view
            name="View"
            style={Object {}}
          >
            <text
              style={Object {}}
            >
              ios-arrow-back
            </text>
          </view>
        </view>
      </view>
      <view
        name="body"
        style={Object {}}
      >
        <view
          name="View"
          style={Object {}}
        >
          <text
            style={Object {}}
          >
            Header
          </text>
        </view>
      </view>
      <view
        name="right"
        style={Object {}}
      >
        <view
          name="View"
          style={
            Object {
              "borderBottomLeftRadius": 2,
              "borderBottomRightRadius": 2,
              "borderTopLeftRadius": 2,
              "borderTopRightRadius": 2,
            }
          }
        >
          <view
            name="View"
            style={Object {}}
          >
            <text
              style={Object {}}
            >
              ios-search
            </text>
          </view>
        </view>
        <view
          name="View"
          style={
            Object {
              "borderBottomLeftRadius": 2,
              "borderBottomRightRadius": 2,
              "borderTopLeftRadius": 2,
              "borderTopRightRadius": 2,
            }
          }
        >
          <view
            name="View"
            style={Object {}}
          >
            <text
              style={Object {}}
            >
              ios-heart
            </text>
          </view>
        </view>
        <view
          name="View"
          style={
            Object {
              "borderBottomLeftRadius": 2,
              "borderBottomRightRadius": 2,
              "borderTopLeftRadius": 2,
              "borderTopRightRadius": 2,
            }
          }
        >
          <view
            name="View"
            style={Object {}}
          >
            <text
              style={Object {}}
            >
              ios-more
            </text>
          </view>
        </view>
      </view>
    </view>
  </view>
</view>
`;

exports[`renders Header with only Title 1`] = `
<view
  name="container"
  style={Object {}}
>
  <view
    name="header"
  >
    <view
      name="statusBar"
    />
    <view
      name="header-other"
      style={Object {}}
    >
      <view
        name="left"
        style={Object {}}
      />
      <view
        name="body"
        style={Object {}}
      >
        <view
          name="View"
          style={Object {}}
        >
          <text
            style={Object {}}
          >
            Header
          </text>
        </view>
      </view>
      <view
        name="right"
        style={Object {}}
      />
    </view>
  </view>
</view>
`;
